maps/src/test/java/org/odk/collect/maps/layers/OfflineMapLayersPickerTest.kt

```diff
@@ -259,7 +283,90 @@ class OfflineMapLayersPickerTest {
         onView(withRecyclerView(R.id.layers).atPositionOnView(1, R.id.radio_button)).check(matches(isChecked()))
     }
 
-    private fun launchFragment(): FragmentScenario<OfflineMapLayersPicker> {
+    @Test
+    fun `clicking the 'add layer' and selecting layers displays the confirmation dialog`() {
+        val scenario = launchOfflineMapLayersPicker()
+
+        uris.add(Uri.parse("blah"))
+        EspressoHelpers.clickOnText(string.add_layer)
+
+        scenario.onFragment {
+            assertThat(
+                it.childFragmentManager.findFragmentByTag(OfflineMapLayersImporter::class.java.name),
+                instanceOf(OfflineMapLayersImporter::class.java)
+            )
+        }
+    }
+
+    @Test
+    fun `clicking the 'add layer' and selecting nothing does not display the confirmation dialog`() {
+        val scenario = launchOfflineMapLayersPicker()
+
+        EspressoHelpers.clickOnText(string.add_layer)
+
+        scenario.onFragment {
+            assertThat(
+                it.childFragmentManager.findFragmentByTag(OfflineMapLayersImporter::class.java.name),
+                equalTo(null)
+            )
+        }
+    }
+
+    @Test
+    fun `progress indicator is displayed during loading layers after receiving new ones`() {
```

Comment:
This test (and the one below) are partly testing `OfflineMapLayersImporter` as well (as they interact with it) which makes the test boundaries in this class a little hard to reason about - early tests verify that `OfflineMapLayersImporter` is "shown", but use the `FragmentManager` to assert on it for instance.

I'd be tempted to test these flows by interacting with the real or a mock `OfflineMapLayersViewModel` instead: calling `importNewLayers` rather than interacting with the UI in `OfflineMapLayersImporter`. An alternative (and smaller change), would be to make this test slightly more "end to end" by asserting that `OfflineMapLayersImporter` is shown by verifying the UI displayed rather than asserting on the `FragmentManager`. This would be far less normal for us though as most of our Fragment tests only test the Fragment itself and not any Fragments they might launch.